community/ndp/messaging-v1/src/main/java/org/neo4j/ndp/messaging/v1/PackStreamMessageFormatV1.java

```diff
@@ -225,10 +225,13 @@ public void flush() throws IOException
         private void packRawMap( Map<String,Object> map ) throws IOException
         {
             packer.packMapHeader( map.size() );
-            for ( Map.Entry<String,Object> entry : map.entrySet() )
+            if(map.size() > 0)
```

Comment:
Nah, it's a silly micro-optimization, I just wanted to avoid having the map create the Iterator (and potentially EntrySet) objects for the loop if it the map was empty, since this is in a really tight part of the serialization code.